x-pack/plugins/ml/public/application/components/ml_inference/components/reindex_with_pipeline.tsx

```diff
@@ -0,0 +1,427 @@
+            {
+              defaultMessage: 'User does not have required permissions to reindex {index}. {error}',
+              values: { error, index: selectedIndex[0].label },
+            }
+          );
+          setCanReindexError(errorMessage);
+        }
+      }
+      if (hasPrivileges !== undefined) {
+        checkPrivileges();
+      }
+    },
+    [hasPrivileges, sourceIndex, toasts, destinationIndex, selectedIndex]
+  );
+
+  useEffect(
+    function checkDestIndexExists() {
+      if (destinationIndex !== undefined && destinationIndex !== '') {
+        debouncedIndexCheck();
+      }
+    },
+    [destinationIndex, debouncedIndexCheck]
+  );
+
+  useEffect(
+    function getIndexOptions() {
+      async function getAllIndices() {
+        const indices = await getIndices();
+        const indexOptions = indices.map((index) => ({ label: index.name }));
+        setOptions(indexOptions);
+      }
+      getAllIndices();
+    },
+    [getIndices]
+  );
+
+  useEffect(
+    function createDiscoverLink() {
+      async function createDataView() {
+        const dataViewCreationResult = await createKibanaDataView(destinationIndex, data.dataViews);
+        if (
+          dataViewCreationResult?.success === true &&
+          dataViewCreationResult?.dataViewId &&
+          showDiscoverLink
+        ) {
+          const url = await generateDiscoverUrl(dataViewCreationResult.dataViewId);
+          setDiscoverLink(url);
+        }
+      }
+      if (reindexingTaskId !== undefined && shouldCreateDataView === true) {
+        createDataView();
+      }
+    },
+    [
+      reindexingTaskId,
+      destinationIndex,
+      data?.dataViews,
+      generateDiscoverUrl,
+      showDiscoverLink,
+      shouldCreateDataView,
+    ]
+  );
+
+  const reindexButton = (
+    <EuiButton
+      onClick={onReindex}
+      disabled={
+        (destinationIndexInvalidMessage !== undefined && selectedIndex.length > 0) ||
+        !canReindex ||
+        destinationIndexExists
+      }
+      size="s"
+    >
+      <FormattedMessage
+        id="xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.review.reindexLabel"
+        defaultMessage="Reindex"
+      />
+    </EuiButton>
+  );
+
+  return (
+    <EuiPanel hasShadow={false} hasBorder={false}>
+      {reindexingTaskId === undefined ? (
+        <EuiFlexGroup direction="column">
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup alignItems="center" gutterSize="xs">
+              <EuiFlexItem grow={false}>
+                <EuiText>
+                  <h4>
+                    <FormattedMessage
+                      id="xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.review.reindexingTitle"
+                      defaultMessage="Reindex with pipeline"
+                    />
+                  </h4>
+                </EuiText>
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <EuiButtonIcon
```

Comment:
Where we link out to the docs, we use a Learn more link in other places (this is from the Calendars page). Can you check with docs which style we should use here. We use the info icon in other places for tooltip style help.

<img width="110" alt="image" src="https://github.com/elastic/kibana/assets/7405507/49c6776a-ac5a-447f-99a9-20a951626983">

Reply:
After chatting with @lcawl, updated to have the link be the `Learn more` text under the heading and the info icon contain some text to briefly explain in-page. cc @peteharverson